file_organizer_v2/src/file_organizer/services/audio/metadata_extractor.py

```
@@ -0,0 +1,353 @@
+            year=int(tag.year) if tag.year else None,
+            track_number=int(tag.track) if tag.track else None,
+            disc_number=int(tag.disc) if tag.disc else None,
+            comment=tag.comment,
+        )
+
+        logger.info(f"Metadata extracted (tinytag): {metadata.duration:.2f}s")
+        return metadata
+
+    def extract_batch(
+        self,
+        audio_paths: List[Union[str, Path]]
```

Ruff (0.14.13)

Use `X | Y` for type annotations

Convert to `X | Y`

(UP007)